### PR TITLE
Mil 115 frondend register

### PR DESCRIPTION
**Pull request by szymonkwak**

### CODE REVIEW

File: client/src/components/SignUp/SignUp.jsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import axios from 'axios';
+import { Card, Box, Typography, Button } from '@mui/material';
+import { useForm } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { useNavigate } from 'react-router-dom';
+import { signUpSchema } from './SignUpSchema';
+import ControllerTextField from './ControllerTextField';
+import SignUpError from './SignUpError';
+
+const SERVER_ADDRESS = 'http://localhost:3000';
```
Comment (CaptainObjective) [inline client/src/components/SignUp/SignUp.jsx]: As discussed please put it into separate file and use to set axios default so we don't have to set it on per request basis

File: client/src/components/SignUp/SignUpSchema.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import * as yup from 'yup';
+import yupLocalePL from 'yup-locale-pl';
+
+yup.setLocale(yupLocalePL);
+
+const MIN_PASSWORD_LENGTH = 5;
+
+// eslint-disable-next-line import/prefer-default-export
```
[the remark this answers is not in the document]
Comment (szymonkwak) [reply client/src/components/SignUp/SignUpSchema.js:8]: done

File: client/src/components/SignUp/ControllerTextField.jsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { TextField } from '@mui/material';
+import { Controller } from 'react-hook-form';
+
+const ControllerTextField = ({ name, label, fieldType, control }) => {
```
[the remark this answers is not in the document]
Comment (szymonkwak) [reply client/src/components/SignUp/ControllerTextField.jsx:5]: done

File: client/src/components/SignUp/SignUp.jsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  });
+
+  const handleSignUp = async (user) => {
+    try {
+      const response = await axios.post(`${SERVER_ADDRESS}/api/auth/sign-up`, {
+        username: user.imię,
+        email: user.email,
+        password: user.confirmPassword
+      });
+      if (response.status === 200) {
```
[the remark this answers is not in the document]
Comment (szymonkwak) [reply client/src/components/SignUp/SignUp.jsx:27]: done

File: client/src/components/SignUp/SignUp.jsx (hunk 1 above)
Comment (szymonkwak) [reply client/src/components/SignUp/SignUp.jsx]: done

**Review by matizaj — COMMENTED** [review]
I'd also add honeypot to register/ login form

**Pull request merged**